### PR TITLE
Update CONTRIBUTING.rst

### PR DESCRIPTION
**Pull request by ltetrel**
Adding instructions to build examples. Not sure what it should be for the main doc though.

### CODE REVIEW

**Comment by kchawla-pi** [comment]
Thanks @ltetrel ! There's a `requirements-build-docs.txt` in the project root that people can install directly if they want all python packages for building docs. Maybe you can mention that.

Also, if you look at issue #2258 , perhaps you can mention the workaround in the issue as well.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
 Nilearn follows the coding conventions used by scikit-learn. `Please read them
 <http://scikit-learn.org/stable/developers/contributing.html#coding-guidelines>`_
 before you start implementing your changes.
+
+Documentation
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
Contributing to the documentation
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+--------------
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
-------------------------------------------------
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+
+To build our documentation, we are using `sphinx <https://www.sphinx-doc.org/en/master/usage/quickstart.html>`_ for the main documentation and `sphinx-gallery <https://sphinx-gallery.github.io/stable/index.html>`_ for the example tutorials.
+If you want to make changes to the example tutorials, please do the following :
+
+1. First, ensure that you have installed sphynx and sphynx gallery
+2. Clone the nilearn repostiory and go to ``nilearn/examples``
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
2. Clone the nilearn repository and go to ``nilearn/examples``
```

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
 <http://scikit-learn.org/stable/developers/contributing.html#coding-guidelines>`_
 before you start implementing your changes.
+
+Documentation
+--------------
+
+To build our documentation, we are using `sphinx <https://www.sphinx-doc.org/en/master/usage/quickstart.html>`_ for the main documentation and `sphinx-gallery <https://sphinx-gallery.github.io/stable/index.html>`_ for the example tutorials.
+If you want to make changes to the example tutorials, please do the following :
+
+1. First, ensure that you have installed sphynx and sphynx gallery
```
Comment (kchawla-pi) [inline CONTRIBUTING.rst]: sphinx and sphinx-gallery

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+2. Clone the nilearn repostiory and go to ``nilearn/examples``
+3. Make your changes using `reStructuredText files <http://docutils.sourceforge.net/rst.html>`_
+4. You can now go to `nilearn/doc` and build the examples locally:
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]: Do the rst codeblock: starts with "::" and indented after.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+
+``python3 -m sphinx -b html -d _build/doctrees . _build/html``
+
+5. Once you are correct with the html outputs in ``nilearn/doc/_build/html/auto_examples/``, you are good to submit the PR!
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
5. Once you are happy with the html outputs in ``nilearn/doc/_build/html/auto_examples/``, you are good to submit the PR!
```

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+
+Documentation
+--------------
+
+To build our documentation, we are using `sphinx <https://www.sphinx-doc.org/en/master/usage/quickstart.html>`_ for the main documentation and `sphinx-gallery <https://sphinx-gallery.github.io/stable/index.html>`_ for the example tutorials.
+If you want to make changes to the example tutorials, please do the following :
+
+1. First, ensure that you have installed sphynx and sphynx gallery
+2. Clone the nilearn repostiory and go to ``nilearn/examples``
+3. Make your changes using `reStructuredText files <http://docutils.sourceforge.net/rst.html>`_
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]: I think that the restructured primer in sphinx is more helpful that the official rst guide.

File: CONTRIBUTING.rst (hunk 3 above)
Comment (kchawla-pi) [reply CONTRIBUTING.rst]:
```rst
2. Fork the Nilearn repository and clone your fork. 
3. Then go to ``nilearn/examples``
```

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+4. You can now go to `nilearn/doc` and build the examples locally:
+
+``python3 -m sphinx -b html -d _build/doctrees . _build/html``
```
Comment (kchawla-pi) [inline CONTRIBUTING.rst]: We use `make html-strict`

File: CONTRIBUTING.rst (hunk 6 above)
Comment (kchawla-pi) [inline CONTRIBUTING.rst]: 7. Visually review the output in ``nilearn/doc/_build/html/auto_examples/``. If all looks well and there were no errors, commit and push the changes.

File: CONTRIBUTING.rst (hunk 4 above)
Comment (emdupre) [inline CONTRIBUTING.rst]:
```suggestion
1. First, ensure that you have installed Sphinx and sphinx-gallery
```

Also, we could probably link these out !

File: CONTRIBUTING.rst (hunk 6 above)
Comment (kchawla-pi) [reply CONTRIBUTING.rst]: 8. You can now open a Pull Request from Nilearn's Pull Request page.

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -68,3 +68,22 @@ Coding guidelines
...
+
+5. Once you are correct with the html outputs in ``nilearn/doc/_build/html/auto_examples/``, you are good to submit the PR!
+
+TIPS : To reduce building time, we suggest you to use the ``filename_pattern`` to build just one specific file.
```
Comment (kchawla-pi) [inline CONTRIBUTING.rst]: For more details about the Fork Clone Push worksflow, `read here <https://guides.github.com/activities/forking/>`_

File: CONTRIBUTING.rst (hunk 8 above)
Comment (ltetrel) [reply CONTRIBUTING.rst]: can you give me the entire command ?

File: CONTRIBUTING.rst (hunk 5 above)
Comment (ltetrel) [reply CONTRIBUTING.rst]: do you have a working example ? I tried few times but it does not seem to work
::
>>my code 
and also
::>>my code

File: CONTRIBUTING.rst (hunk 8 above)
Comment (GaelVaroquaux) [reply CONTRIBUTING.rst]: The command is "make html-strict".

I think that I'll merge anyhow, as this is better than what we had before.

**Comment by kchawla-pi** [comment]
Wait I can make the changes and push them then merge tomorrow

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -68,3 +68,27 @@ Coding guidelines
...
+To build our documentation, we are using `sphinx <https://www.sphinx-doc.org/en/master/usage/quickstart.html>`_ for the main documentation and `sphinx-gallery <https://sphinx-gallery.github.io/stable/index.html>`_ for the example tutorials.
+If you want to make changes to the example tutorials, please do the following :
+
+1. First, ensure that you have installed sphinx and sphinx-gallery. You can install the requirements using ``nilearn/requirements-build-docs.txt``.
+2. Fork the Nilearn repository and clone your fork.
+3. Then go to ``nilearn/examples``
+4. Make your changes using `reStructuredText files <https://www.sphinx-doc.org/en/2.0/usage/restructuredtext/basics.html>`_
+5. You can now go to `nilearn/doc` and build the examples locally:
+
+::      python3 -m sphinx -b html -d _build/doctrees . _build/html
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
       python3 -m sphinx -b html -d _build/doctrees . _build/html
```

File: CONTRIBUTING.rst [hunk 11]
```diff
@@ -68,3 +68,27 @@ Coding guidelines
...
+-------------------------------------------------
+
+To build our documentation, we are using `sphinx <https://www.sphinx-doc.org/en/master/usage/quickstart.html>`_ for the main documentation and `sphinx-gallery <https://sphinx-gallery.github.io/stable/index.html>`_ for the example tutorials.
+If you want to make changes to the example tutorials, please do the following :
+
+1. First, ensure that you have installed sphinx and sphinx-gallery. You can install the requirements using ``nilearn/requirements-build-docs.txt``.
+2. Fork the Nilearn repository and clone your fork.
+3. Then go to ``nilearn/examples``
+4. Make your changes using `reStructuredText files <https://www.sphinx-doc.org/en/2.0/usage/restructuredtext/basics.html>`_
+5. You can now go to `nilearn/doc` and build the examples locally:
```
Comment (GaelVaroquaux) [inline CONTRIBUTING.rst]:
```suggestion
5. You can now go to `nilearn/doc` and build the examples locally::
```

**Pull request merged**